{{ cookiecutter.github_repo_name }}/src/{{ cookiecutter.package_dir_name }}/calculator.py

```diff
@@ -2,5 +2,30 @@
 
 
 def dot_product(a, b):
-    """Calculate the dot product of two vectors."""
-    return np.dot(a, b)
+    """Compute the dot product of two vectors of any size.
+
+    Ensure that the inputs, a and b, are of the same size.
+    The supported types are "array_like" objects, which can
+    be converted to a NumPy array. Examples include lists and tuples.
+
+    Parameters
+    ----------
+    a : array_like
```

Comment:
numpy uses the term `array_like`. Ref: https://numpy.org/doc/stable/reference/generated/numpy.dot.html